x-pack/plugins/security_solution/public/overview/components/detection_response/recent_cases_table/use_recent_cases_items.ts

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import moment from 'moment';
+import { useState, useEffect, useCallback } from 'react';
+
+import { CaseStatuses } from '@kbn/cases-plugin/common';
+import { AllCases } from '@kbn/cases-plugin/common/ui';
+
+import { APP_ID } from '../../../../../common/constants';
+import { useGlobalTime } from '../../../../common/containers/use_global_time';
+import { useKibana } from '../../../../common/lib/kibana';
+import { addError } from '../../../../common/store/app/actions';
+
+export interface RecentCaseItem {
+  name: string;
+  note: string;
+  createdAt: string;
+  createdBy: string;
+  status: CaseStatuses;
+  id: string;
+}
+
+export interface UseRecentlyOpenedCasesProps {
+  skip: boolean;
+}
+
+export type UseRecentlyOpenedCases = (props: UseRecentlyOpenedCasesProps) => {
+  items: RecentCaseItem[];
+  isLoading: boolean;
+  updatedAt: number;
+};
+
+export const useRecentlyOpenedCases: UseRecentlyOpenedCases = ({ skip }) => {
+  const {
+    services: { cases },
+  } = useKibana();
+  const { to, from } = useGlobalTime();
+  const [isLoading, setIsLoading] = useState(true);
+  const [updatedAt, setUpdatedAt] = useState(Date.now());
+  const [items, setItems] = useState<RecentCaseItem[]>([]);
+
+  // This is a known issue of cases api, it doesn't accept date time format atm
+  // Once they fix this problem we can remove this two lines
+  // [TO-DO] #130979
+  const fromDate = moment(from).format('YYYY-MM-DD');
+  const toDate = moment(to).format('YYYY-MM-DD');
+
+  const fetchCases = useCallback(async () => {
+    try {
+      const casesResponse = await cases.api.cases.find({
+        from: fromDate,
+        to: toDate,
+        owner: APP_ID,
+        sortField: 'create_at',
+        sortOrder: 'desc',
+        page: 1,
+        perPage: 4,
+      });
+
+      setItems(parseRecentCases(casesResponse as unknown as AllCases));
+    } catch (err) {
+      addError(err);
```

Comment:
Ive been wondering about how we error handle in the app, In some cases we use addError, Ive also noticed we sometimes surface no error, but return an empty object or list. Whats preferred? or both?